lib/RequestLogger.js

```diff
@@ -222,6 +222,19 @@ class RequestLogger {
     }
 
     /**
+     *  This function creates a new RequestLogger object from a ModuleLogger
+     *  already existing in order to have the submodule field from the
+     *  ModuleLogger.
+     *
+     *	@param {ModuleLogger} ml - The base ModuleLogger
+     *	@return {RequestLogger} - The new RequestLogger created
+     *				  with the ModuleLogger
+     */
+    scope(ml) {
```

Comment:
Creating a new RequestLogger effectively changes the request UID. That's why this is out of the question, messing with our request tracking capabilities. Hence my suggestion to _wrap_ a requestLogger, much like the EndLogger. You  do not need to re-write the RequestLogger; only to wrap it and add what you need.